### PR TITLE
make bq_dataset configurable through bq_dataset.txt

### PR DESCRIPTION
**Pull request by hanlu09205**
Add bq_dataset.txt for both staging and prod. 
Udpate gke.sh to take in bq_dataset from the test file. 
Udpate README

### CODE REVIEW

File: deployment/README.md [hunk 1]
```diff
@@ -18,6 +18,9 @@ Create a Google Cloud Platform (GCP) project and run the following command where
 To build a new image, run the command below. Tag number should be larger than any existing one. 
 gcloud builds submit --tag gcr.io/datcom-mixer/go-grpc-mixer:<TAG> 
 
+If you need to point to a different Bigtable table or BigQuery dataset, update the corresponding bt_table.txt or bq_dataset.txt.
+Both of them have to version, prefixed with prod_ or staging_.
```
Comment (pradh) [inline deployment/README.md]: have two versions

File: deployment/gke.sh [hunk 2]
```diff
@@ -51,11 +51,16 @@ else
 fi
 
 # Set BQ dataset
-if [ "$PROJECT_ID" == "datcom-mixer-staging" ]; then
-  perl -i -pe's/BQ_DATASET/google.com:datcom-store-dev.dc_v3_staging_clustered/g' deployment.yaml
-else 
-  perl -i -pe's/BQ_DATASET/google.com:datcom-store-dev.dc_v3_clustered/g' deployment.yaml
+if [ "$PROJECT_ID" == "datcom-mixer" ]; then
+  bq_dataset_input_file="./prod_bq_dataset.txt"
```
Comment (pradh) [inline deployment/gke.sh]: ./ is redundant?

File: deployment/gke.sh [hunk 3]
```diff
@@ -51,11 +51,16 @@ else
...
+else
+  bq_dataset_input_file="./staging_bq_dataset.txt"
 fi
+while read -r line; do
+  BQ_DATASET=$line
+done < "$bq_dataset_input_file"
```
Comment (pradh) [inline deployment/gke.sh]: why not?

BQ_DATASET=$(cat $bq_dataset_input_file)

File: deployment/gke.sh [hunk 4]
```diff
@@ -51,11 +51,16 @@ else
...
+export BQ_DATASET
+perl -i -pe's/BQ_DATASET/$ENV{BQ_DATASET}/g' deployment.yaml
```
Comment (pradh) [inline deployment/gke.sh]: I'm curious why perl was used, long time no see :)

this is fine for consistency, but an alternative would be to do:

    sed -i "s/BQ_DATASET/$BQ_DATASET/g" deployment.yaml

which doesn't need "export BQ_DATASET".

Comment (hanlu09205) [reply deployment/gke.sh]: I got sed: 1: "deployment.yaml": extra characters at the end of d command
Maybe if we use sed, different OS will respond different, vs perl.

File: deployment/gke.sh (hunk 3 above)
Comment (hanlu09205) [reply deployment/gke.sh]: Thanks for the suggestion.

Comment (hanlu09205) [reply deployment/gke.sh]: I also updated how BT_TABLE get assigned.

**Comment by hanlu09205** [comment]
PTAL

**Pull request merged**